app/models/binary/DataSetService.scala

```diff
@@ -20,35 +16,40 @@ object DataSetService extends FoxImplicits with LazyLogging {
 
   val system = Akka.system(play.api.Play.current)
 
-  def updateTeams(dataSet: DataSet, teams: List[ObjectId])(implicit ctx: DBAccessContext) =
-    DataSetDAO.updateTeams(dataSet.name, teams)
-
-  def update(dataSet: DataSet, description: Option[String], displayName: Option[String], isPublic: Boolean)(implicit ctx: DBAccessContext) =
-    DataSetDAO.update(dataSet.name, description, displayName, isPublic)
-
   def isProperDataSetName(name: String): Boolean =
     name.matches("[A-Za-z0-9_\\-]*")
 
-  def checkIfNewDataSetName(name: String)(implicit ctx: DBAccessContext): Fox[Boolean] =
-    findDataSource(name)(GlobalAccessContext).reverse
+  def assertNewDataSetName(name: String)(implicit ctx: DBAccessContext): Fox[Boolean] =
+    DataSetSQLDAO.findOneByName(name)(GlobalAccessContext).reverse
 
   def createDataSet(
                      name: String,
                      dataStore: DataStoreInfo,
                      owningOrganization: String,
                      dataSource: InboxDataSource,
                      isActive: Boolean = false
-                     )(implicit ctx: DBAccessContext) = {
+                     ) = {
+    implicit val ctx = GlobalAccessContext
```

Comment:
Did you change this because `ctx` was never explicitly set?

Reply:
the passed ctx was never used since the only call that would have captured it was the DataSetDAO.insert, which was instead called with the GlobalAccessContext.
Now, I use the GlobalAccessContext for all three, so to reduce the redundancy I’ll declare it implicit in this scope, and discard the passed one entirely.